### PR TITLE
lib: misc: Add BinTools

### PR DESCRIPTION
**Pull request by dnltz**
Add a tool to handle binary software images. Right now, this tool can only
initialize RAMs.

### CODE REVIEW

File: lib/src/main/scala/spinal/lib/misc/BinTools.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import spinal.core.{Data, Mem}
+
+object BinTools {
+  def initRam[T <: Data](ram : Mem[T], onChipRamBinFile : String): Unit ={
+    val initContent = Array.fill[BigInt](ram.wordCount)(0)
+    val readTmp = Array.fill[Byte](4)(0)
+    val initFile = new FileInputStream(onChipRamBinFile)
+    for ((e, i) <- initContent.zipWithIndex) {
+      if (initFile.read(readTmp) == 0) {
+        initContent(i) = 13
```
[the remark this answers is not in the document]
Comment (dnltz) [reply lib/src/main/scala/spinal/lib/misc/BinTools.scala]: Nops, but maybe illegal ops would be better to provoke an exception.

File: lib/src/main/scala/spinal/lib/misc/BinTools.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package spinal.lib.misc
+
+import java.io.FileInputStream
+import spinal.core.{Data, Mem}
+
+object BinTools {
+  def initRam[T <: Data](ram : Mem[T], onChipRamBinFile : String): Unit ={
+    val initContent = Array.fill[BigInt](ram.wordCount)(0)
+    val readTmp = Array.fill[Byte](4)(0)
```
[the remark this answers is not in the document]
Comment (dnltz) [reply lib/src/main/scala/spinal/lib/misc/BinTools.scala]: +1

File: lib/src/main/scala/spinal/lib/misc/BinTools.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dolu1990) [reply lib/src/main/scala/spinal/lib/misc/BinTools.scala]: Ahhh i got it now.
Yes, let's go for an exception. Or a padding option ?

[the remark this answers is not in the document]
Comment (dnltz) [reply lib/src/main/scala/spinal/lib/misc/BinTools.scala]: `Array.fill[Byte](ram.width / 8)(0)` this Array is already initialized with 0 and this if case is unnecessary.

**Comment by Dolu1990** [comment]
Thanks :D

**Pull request merged**